UI/window-basic-main.cpp

```diff
@@ -6100,21 +6100,21 @@ void OBSBasic::on_streamButton_clicked()
 		bool confirm = config_get_bool(GetGlobalConfig(), "BasicWindow",
 					       "WarnBeforeStartingStream");
 
-		obs_data_t *settings = obs_service_get_settings(service);
-		bool bwtest = obs_data_get_bool(settings, "bwtest");
-		obs_data_release(settings);
-
-		if (bwtest && isVisible()) {
-			QMessageBox::StandardButton button =
-				OBSMessageBox::question(
-					this, QTStr("ConfirmBWTest.Title"),
-					QTStr("ConfirmBWTest.Text"));
+		auto bwAction =
+			UIValidation::BandwidthModeConfirmation(this, service);
+		switch (bwAction) {
+		case StreamSettingsAction::ContinueStream:
+			break;
+		case StreamSettingsAction::OpenSettingsStream:
+			on_action_Settings_triggered(1);
```

Comment:
oh I see. Further on enum usage, I think take this parsing out of Main Window and just send settings the enum to figure itself out?